File: subscribes/service_csv.py
import redis
import pandas as pd

server = redis.Redis(host='localhost', port=6379, db=0)
pubsub = server.pubsub()

pubsub.subscribe('topic_create_file')

for b in pubsub.listen():
    data = b['data']
    if data != 1:
        df = pd.read_json(data)
        df.to_csv('./files/file_python.csv', encoding='utf-8', index=False)
        print(f"File python CSV\n")

File: subscribes/service_csv.go
package main

import (
	"github.com/gomodule/redigo/redis"
	"fmt"
	"encoding/csv"
    "encoding/json"
    "os"
)

type Application struct {
    Brand string
    Price string
}

func main() {	
	c, err := redis.Dial("tcp", ":6379")
	if err != nil {
		// handle error
	}
	
	psc := redis.PubSubConn{Conn: c}
	psc.Subscribe("topic_create_file")

	for {
		switch v := psc.Receive().(type) {
			case redis.Message:
				var jsonData []Application
				err = json.Unmarshal([]byte(v.Data), &jsonData)
				if err != nil {
					fmt.Println(err)
				}

				csvFile, err := os.Create("./files/file_go.csv")
				if err != nil {
					fmt.Println(err)
				}
				defer csvFile.Close()

				writer := csv.NewWriter(csvFile)

				for _, usance := range jsonData {
					var row []string
					row = append(row, usance.Brand)
					row = append(row, usance.Price)
					writer.Write(row)
				}
				writer.Flush()
				fmt.Println("File go CSV")
		}
	}
	defer c.Close()
}
